prow/started_and_finished.md

```diff
@@ -0,0 +1,123 @@
+
+Related Issues:
+
+1. #3412: What is the origin and porous for the fields in these files?
+2. #111000: This isn't a source of truth and prow/pod/gcs are not in sync
+3. #10699: Unify *.json structures, was partially covered as part of #10703
+
+## Format Source of Truth
+There has not been a consistent source of truth for the format of these two files, which has caused issues. From discussion in the community it seems that the the [TestGrid job definition](https://github.com/GoogleCloudPlatform/testgrid/blob/master/metadata/job.go).
+
+## Current Standards
+There are currently different flavors of data format depending on if the job is Bootstrap or PodUtils. Ex of differences:
+```
+Bootstrapped PR (finished): "revision": "v1.20.0-alpha.0.261+06ea384605f172"
+Decorated PR (finished): "revision":"5dd9241d43f256984358354d1fec468f274f9ac4"
+```
+
+[`Started.json` *Bootstrap*](https://github.com/kubernetes/test-infra/blob/a1a207e4cd847671f0a53553c664e24d26c9cdf7/jenkins/bootstrap.py#L315)
+|Fields|Content|
+|---|---|
+|node|This is the first element in the hostname using socket.gethostname split by '.'|
+|pull|The SHA linked with the 'main' repo within 'repos'|
+|repo-version|"unknown" if no 'repos' otherwise read from local 'version' file (e2e tests use this path) otherwise execute version script if 'hack/lib/version.sh exists|
+|timestamp|epoch time|
+|repos|comes from --repos= arg|
+|version|exact same as `repo-version`|
+*Ex*
+```
+{
+  "node": "0790211c-cacb-11ea-a4b9-4a19d9b965b2",
+  "pull": "master:5a529aa3a0dd3a050c5302329681e871ef6c162e,93063:c25e430df7771a96c9a004d8500473a4f2ef55d3",
+  "repo-version": "v1.20.0-alpha.0.261+06ea384605f172",
+  "timestamp": 1595278460,
+  "repos": {
+    "k8s.io/kubernetes": "master:5a529aa3a0dd3a050c5302329681e871ef6c162e,93063:c25e430df7771a96c9a004d8500473a4f2ef55d3",
+    "k8s.io/release": "master"
+  },
+  "version": "v1.20.0-alpha.0.261+06ea384605f172"
+}
+```
+
+[`Finished.json` *Bootstrap*](https://github.com/kubernetes/test-infra/blob/1a958b0c2b6ddbb813bf6d23fe6b5714e9812e38/jenkins/bootstrap.py#L521)
+|Fields|Content|
+|---|---|
+|timestamp|epoch|
+|passed|bool (job success)|
+|version|If version is in metadata, set from metadata same as job-version|
+|result|'SUCCESS' or "FAILURE' depending on passed|
+|job-version (dep)|If not existing and not 'unknown'... from metadata, try 'job-version' then 'version'|
+|metadata|exact same as `repo-version`|
+|- repo-commit|Git rev-parse HEAD (for k8s)|
+|- repos|Same as started 'comes from --repo= args'|
+|- infra-commit|Git rev-parse HEAD (for test-infra)|
+|- repo|main repo for job|
+|- job-version| Same as job version from above|
+|- revision| Same as job-version|
```

Comment:
The notation with `-` for sub-members looks a bit weird, maybe we should use jsonpath and also format the fields with monospace, e.g.`.metadata.repos` etc

Reply:
Like it, done